### DISCUSSION TITLE
mvcpaging-2-0/

### DISCUSSION THREAD

**Discussion by martijnboland** [martijnboland/blog#45] [Blog comments]
# mvcpaging-2-0/

https://blogs.taiga.nl/martijn/posts/mvcpaging-2-0/

**Comment by disqus-comments-migrator** [comment]
<em>AM commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at april 24 2012, 10:08</em>

---

<p>Hello, I just started using this library....great work.  Is there back button support that I am missing somewhere?  If it's not included, would this be something that could be incorporated later on?  Great work.</p>

**Comment by disqus-comments-migrator** [comment]
<em>martijn commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at april 25 2012, 02:03</em>

---

<p>What do you mean with back button support? Ajax scenarios?</p>

**Comment by disqus-comments-migrator** [comment]
<em>chlebta commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at mei 19 2012, 08:15</em>

---

<p>Please can you make a tuto to how to use it didn't get the point in your old post :(</p>

**Comment by disqus-comments-migrator** [comment]
<em>JLR commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at mei 28 2012, 07:10</em>

---

<p>Hi Martin,</p><p>I've been using your MVCPager for some time, great work, thanks.</p><p>I do have a small problem which I hope you can help me with, the domain name is missing in the link within the pager whenever I don't use the deepest url on my search controller.</p><p>For example this page is fine</p><p><a href="http://www.independentmountain.com/ski-search/france/chamonix-mont-blanc/chamonix/catered/sleepsany" rel="nofollow noopener" title="http://www.independentmountain.com/ski-search/france/chamonix-mont-blanc/chamonix/catered/sleepsany">http://www.independentmountain.com/ski-search/france/chamonix-mont-blanc/chamonix/catered/sleepsany</a></p><p>However I have landing pages for each parameter in the above path (linked to within individual results) and hance use pages such as this</p><p><a href="http://www.independentmountain.com/ski-search/france/" rel="nofollow noopener" title="http://www.independentmountain.com/ski-search/france/">http://www.independentmountain.com/ski-search/france/</a></p><p>On the latter page you'll see that the pager links are not fully qualified and hence broken, any thoughts on how I can fix this?</p><p>TIA.</p><p>JLR</p>

**Comment by disqus-comments-migrator** [comment]
<em>JLR commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at mei 28 2012, 08:15</em>

---

<p>I've fixed the above issue.  I had all my url parameters using UrlParameter.Optional in my global.asax.  Now using default values and the pager is working as expected.</p><p>Hope this might help somebody else.</p><p>JLR</p>

**Comment by disqus-comments-migrator** [comment]
<em>Prathiba commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at juni 09 2012, 07:54</em>

---

<p>Hi Martin,<br>I am using your MVCPaging control.<br>In one our page as <br>@Html.Pager(Model.PageSize, Model.PageNumber, Model.TotalItemCount, new AjaxOptions { UpdateTargetId = "gridcontainer" }).Options(o =&gt; o.Action("MyActionForAjaxPage"))<br>But I face an issue like I have given the page size as 1 for testing purpose<br>I get some 2 pages, When I navigate through pages, I am able to navigate.<br>Then I logout the application and login is another user, who say for example has 3 pages<br>when I navigat through pages it tries to show the previous user's page, when I try to go to page 3, it show me as only 2 pages are there and that to previous users page is shown.</p><p>I don't have any cache mechanism in my application, Even I felt it may be due to temp data, I deleted the IE history, but even then I am gettting the same issue consistently, Can you please tell me If I am missing something,</p><p>Even When I try to navigate to 2nd or 3rd page the breakpoint in the action method is not being hit.</p><p>Please help me out as this is a critical issue we are facing.</p>

**Comment by disqus-comments-migrator** [comment]
<em>martijn commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at juni 09 2012, 09:23</em>

---

<p>Could you please try a different browser than IE? IE is known to cache AJAX results very aggresively. <br>You might also try to disable caching via an attribute on the AJAX action: <br>[OutputCache(NoStore = true, Duration = 0, VaryByParam = "*")].</p>

**Comment by disqus-comments-migrator** [comment]
<em>Prathiba commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at juni 10 2012, 12:52</em>

---

<p>Thanks a lot it is working perfectly now , after giving the specified option in the Ajax action, Also in firefox even without that cache disable option it is working fine</p>

**Comment by disqus-comments-migrator** [comment]
<em>Neptune commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at juli 09 2012, 04:26</em>

---

<p>Hello there and many thanks to your MvcPaging!!</p><p>I have a question, I'm using partiall views for my pages, so I need to add to my partial view the option "Action". This is fine, but.. Although I see the url is http://blahblah/blah?page=something at my link hover, when I actually click the link to go to the next page, the route value "page" disappears from my link ( the results are fine )<br>How do I add it back??</p><p>Thanks in advance!</p>

**Comment by disqus-comments-migrator** [comment]
<em>Neptune commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at juli 09 2012, 07:16</em>

---

<p>Uhm.. sorry for my question, I closed Visual Studio, I opened it again and everything worked!</p>

**Comment by disqus-comments-migrator** [comment]
<em>po commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at september 12 2012, 12:25</em>

---

<p>I have an error as below.<br>Page Number cannot be below 1.<br>what should i do now?</p>

**Comment by disqus-comments-migrator** [comment]
<em>Hassan commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at september 25 2012, 02:40</em>

---

<p>Hello</p><p>Great library ! Really easy to install and make it working</p><p>I have a question :</p><p>I have a generated survey form with 10 Questions, I want to paginate it.<br>All works but how can I assign to the Next and Previous button a submission of the form ?</p><p>Thanks !</p>

**Comment by disqus-comments-migrator** [comment]
<em>martijn commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at september 25 2012, 05:16</em>

---

<p>@Hassan</p><p>I don't know if that's easy. I'd probably try something like (javascript/jQuery):<br>- intercept the pager click (link)<br>- extract the page number from the pager link and put that in a hidden field in the form<br>- submit the form<br>- return false, so the original pager click is discarded</p>

**Comment by disqus-comments-migrator** [comment]
<em>Hassan commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at september 25 2012, 09:18</em>

---

<p>Thanks this is a great idea ! Will try to implement it :-)</p>

**Comment by disqus-comments-migrator** [comment]
<em>Hassan commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at oktober 05 2012, 09:45</em>

---

<p>Hello</p><p>Your solution for the paging and submit works like a charm ! I have another issue, I tried to find a solution but I've not succeeded :</p><p>I want to "customize" the paging. I explain :<br>Example with a paging of 2 elements / page<br>E1 - true -&gt; Page 1<br>E2 - true -&gt; Page 1</p><p>E3 - true -&gt; Page 2<br>E4 - false -&gt; Page 2<br>E5 - true -&gt; Page 2</p><p>E6 - true -&gt; Page 3</p><p>Each element with IsVisible = false, will be inserted to the page but will not be counted</p><p>If you have any Idea, thank you very much again</p>

**Comment by disqus-comments-migrator** [comment]
<em>Nikolas commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at oktober 14 2012, 01:45</em>

---

<p>Hi</p><p>I am passing a routevaluedictionary to my Pager</p><p>@Html.Pager(Model.Problems.PageSize, Model.Problems.PageNumber, Model.Problems.TotalItemCount).Options(o =&gt; o.RouteValues(tRVD).Action("Search"))</p><p>The problem is that my routevaluedictionary contains a list  of (PostedSpecialties.PostedSpecialty) values (one key -  multiple values)<br>and Pager returns URL</p><p> PostedSpecialties PostedSpecialty = 2,3,4</p><p>instead of</p><p> PostedSpecialties.PostedSpecialty=2&amp;PostedSpecialties.PostedSpecialty=3&amp;PostedSpecialties. PostedSpecialty=4</p><p>Is there any way to solve this?</p>

**Comment by disqus-comments-migrator** [comment]
<em>martijn commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at oktober 14 2012, 06:21</em>

---

<p>Nikolas, <br>I don't know if this could be solved easily. What happens if you make an ActionLink with the same routevalues dictionary?</p>

**Comment by disqus-comments-migrator** [comment]
<em>tryhp3 commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at oktober 28 2012, 07:03</em>

---

<p>Bootstrap demo is great.  Is possible implement MvcPaging for Zurb Fundaiton <a href="http://foundation.zurb.com/docs/navigation.php#pagEx" rel="nofollow noopener" title="http://foundation.zurb.com/docs/navigation.php#pagEx">http://foundation.zurb.com/docs/navigation.php#pagEx</a> ?</p>

**Comment by disqus-comments-migrator** [comment]
<em>martijn commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at oktober 29 2012, 01:42</em>

---

<p>@tryhp3: I think it will take very little to convert the bootstrap example to something that works with Zurb foundation. They're the same, conceptually and you'll probably only have to change some css class names in the DisplayTemplate.</p>

**Comment by disqus-comments-migrator** [comment]
<em>tryhp3 commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at oktober 30 2012, 03:42</em>

---

<p>I want to change the name of the parameter 'page'. For example Controler/Action?page=1 to Controler/Action?PageNumber=1. Is it possible?</p>

**Comment by disqus-comments-migrator** [comment]
<em>tryhp3 commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at oktober 30 2012, 03:55</em>

---

<p>I have additional quesiton. I want use RouteValues(object routeValues) for my Filter model:<br>public class ShopFilter<br> {<br>        public string ProductCode { get; set; }<br>        public int[] SelectedSuppliers { get; set; } //list of checkboxes<br> }   <br>but generated url is wrong:<br>?ProductCode=test&amp;SelectedSuppliers=System.Int32%5B%5D&amp;page=1</p><p>Is possible serialize int[] ?<br>should be:<br>?ProductCode=test&amp;SelectedSuppliers=1&amp;SelectedSuppliers=5&amp;page=1 etc.</p>

**Comment by disqus-comments-migrator** [comment]
<em>martijn commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at oktober 30 2012, 02:10</em>

---

<p>@tryhp3</p><p>-  Yes, you can change the page parameter name with adding the option PageRouteValueKey = "PageNumber"</p><p>-  Unfortunately, this is not possible to use an array of int's with a routevalue dictionary, see <a href="http://stackoverflow.com/questions/6395290/how-may-i-add-integer-list-to-route" rel="nofollow noopener" title="http://stackoverflow.com/questions/6395290/how-may-i-add-integer-list-to-route">http://stackoverflow.com/questions/6395290/how-may-i-add-integer-list-to-route</a></p>

**Comment by disqus-comments-migrator** [comment]
<em>Roger Frehen commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at oktober 30 2012, 06:59</em>

---

<p>Hello Martijn,</p><p>To be able to use your MvcPaging in a base view (for example, a general index page in a rather simple crud application), it would be good to add covariance to IPagedList. Therefore, two things need to be changed:<br>1. The parameter T in the interface IPagedList needs to be marked as covariant, so it will be IPagedList<br>2. Because IList doesn't support covariance either, IPagedList can't inherit from IList anymore. I solved this by having it inherit form IEnumerable instead.</p><p>My question to you is: do you see any problem in doing the latter?</p>

**Comment by disqus-comments-migrator** [comment]
<em>martijn commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at oktober 30 2012, 08:13</em>

---

<p>@Roger</p><p>I don't think there is a problem doing this. Could you perhaps provide a pull request? The source is located at <a href="https://github.com/martijnboland/MvcPaging" rel="nofollow noopener" title="https://github.com/martijnboland/MvcPaging">https://github.com/martijnboland/MvcPaging</a>. Oh, and we use tabs for indentation :)</p>

**Comment by disqus-comments-migrator** [comment]
<em>tryhp3 commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at oktober 31 2012, 01:17</em>

---

<p>Regarding to my question about array of int's. Is possible to use this solution (or similar) - <a href="http://stackoverflow.com/questions/6396136/how-do-i-pass-int-array-to-routevaluedictionary" rel="nofollow noopener" title="http://stackoverflow.com/questions/6396136/how-do-i-pass-int-array-to-routevaluedictionary">http://stackoverflow.com/questions/6396136/how-do-i-pass-int-array-to-routevaluedictionary</a> in your pager?</p>

**Comment by disqus-comments-migrator** [comment]
<em>pty commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at november 04 2012, 11:18</em>

---

<p>Can I do a redirect to another controller? .Action("action","controler") ?</p>

**Reply by disqus-comments-migrator** [reply]
<em>martijn commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at november 05 2012, 03:06, in reply to pty's [comment](https://github.com/martijnboland/blog/discussions/45#discussioncomment-8952938)</em>

---

<p>@pty: You can add a route value to specify a different controller:<br>.Action("action").AddRouteValue("controller", "MyController")</p>

**Comment by disqus-comments-migrator** [comment]
<em>Meysam commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at november 23 2012, 07:04</em>

---

<p>Thanks for great work. <br>why when using Ajax and bootstrap together Ajax functionality doesn't work?</p>
